[WIP] Mid-round antag event - "Syndicate Battle Cruiser"

EgorDinamit opened the pull request:
## About The Pull Request

**Adds an event that spawns syndicate battle cruiser, with 2 smaller shuttles in its bay.** 

Cruiser itself is powered by supermatter engine.
The event can trigger only after 1 hour and only if the server has more than 40 active players.
Cruiser contains 13 "ghost-role spawners" that work similarly to pirates' event, though for the cruiser to actually spawn you will need at least 6 candidates first.

The armoury contains 1 mauler, 1 bag of C4 and X4, 2 grenadier's belts, a lot of other explosives and a good amount of ballistic weapons for at least 11 people.
This PR also adds energy crowbar unique to engineers(2) that spawn on the cruiser and upgrades energy wrench by a bit.

- [ ] Shuttles, that actually spawn on a cruiser once the event triggers.
- [ ] Functioning UI on cruiser's navigation console.
- [ ] Functioning objectives other than "protect SM on the cruiser".
- [ ] Some sort of response in case more than 50% of the crew dies while an event is active
_Probably just a forced shuttle call or ERT spawn(not on centcom z-level though)._

## Why It's Good For The Game

First of all, it makes your favourite greenshifts, or very long rounds, end up in a very interesting manner if you are lucky enough. That means the end of 2+ hours shifts, unless the crew is, in fact, ready to fight.
In my opinion, antag variety is a perfect thing to have in ss13 and not only that, but unlike other late events, such as lone op or space ninja, it gives a fun event for the whole server, and if your local crew on the station can't destroy the cruiser with advanced weaponry you MUST have at 1-hour mark, maybe at least 13 syndicate operatives will have fun.
![crowbar](https://user-images.githubusercontent.com/53223414/79634358-ff73e500-817a-11ea-9d0b-5289fae93cb1.gif)

![syndicate_starfury-1](https://user-images.githubusercontent.com/53223414/79634367-0bf83d80-817b-11ea-9272-bb9c1be41af8.png)

## Changelog
:cl: EgorDinamit & WJohn
add: Added new mid-round antagonist event - Syndicate Battle Cruiser.
add: Security hud icons for syndicate jobs.
add: Energy crowbar!
tweak: Energy wrench damage and tool speed buffed.
imageadd: Added icon for energy crowbar.
/:cl:

EgorDinamit committed:
Starfury Battle Ship: Main Commit

Battleship added.
Syndicate SM subtype added.
Added Event for this ship to spawn.
Added Antag datum for crew that spawns on this ship.
Ghost Roles are given antag status and HUD.
Updated "Starfury BC" decals.

EgorDinamit committed:
Update

EgorDinamit committed:
Update syndicate_starfury.dmm

EgorDinamit committed:
Le great update.

EgorDinamit committed:
Woah, another big update.

EgorDinamit committed:
Le update, changing some docking ports

EgorDinamit committed:
deadchat_broadcast

EgorDinamit committed:
Removes template map and dmm-tools file.

I used it while map was heavily worked on.

EgorDinamit committed:
Merge branch 'master' into syndicatebattleship

ATH1909 commented:
https://www.youtube.com/watch?v=sTAIxU_geYo

EgorDinamit committed:
IDs cleanup

EgorDinamit committed:
Maps converted to TGM

EgorDinamit committed:
Merge branch 'syndicatebattleship' of https://github.com/EgorDinamit/tgstation into syndicatebattleship

EgorDinamit committed:
Removing unused objectives, Setting normal values to event.

Tlaltecuhtli commented:
cool a big chunk of metal to throw ttvs at

EgorDinamit commented:
@Tlaltecuhtli not that I see anyone using TTVs against nukies too often though.
This ship gets shields too, so good luck bombing, though you CAN do it with at least 2 maxcaps.

Pumpkinoe commented:
looks awesome, but, why iz the syndicate stealing supermatter slivers if they have their own, completely functioning one?

TiviPlus commented:
Witness the powah of this fully operation battlestation

minichibis commented:
There is that whole antag freeze thing but this does seem at the very least worth a testmerge to see how it would change the dynamic of the average round.

I will say from the map that the ship does feel a bit rectangular inthe middle but thats from a birds eye view and not a ground level view.

itseasytosee commented:
so a mini-station to fight the big station.
I don't like the lore implications that the syndicate has access to a supermarket for the same reasons I dislike the laser gun implant and stealing the hypospray.

TheMidnightRose commented:
> There is that whole antag freeze thing but this does seem at the very least worth a testmerge to see how it would change the dynamic of the average round.
> 

Antag Freeze is over. Offstation content is currently frozen.

minichibis commented:
ohhh I see

I was reading through a ton of old commits a while ago

EgorDinamit commented:
Perhaps that's why syndicate is stealing supermatter silver - to create their own crystal, using reverse-engineering.
As I mentioned in PR, bay(middle) will have 2 shuttles in it once event triggers, so rn it might feel empty.

bobbahbrown commented on code/game/gamemodes/objective.dm:
```diff
@@ -1137,3 +1137,13 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+
+/datum/objective/syndicatesupermatter
+	name = "keep SM intact"
+	explanation_text = "Keep supermatter engine onboard the cruiser intact."
```
```suggestion
	explanation_text = "Keep the supermatter engine onboard the cruiser intact."
```

bobbahbrown commented on code/game/gamemodes/objective.dm:
```diff
@@ -1137,3 +1137,13 @@ GLOBAL_LIST_EMPTY(possible_items_special)
+	explanation_text = "Keep supermatter engine onboard the cruiser intact."
+
+/datum/objective/syndicatesupermatter/check_completion()
+	if(GLOB.syndicate_supermatter_engine)
+		return TRUE
+	else
+		return FALSE
```
```suggestion
	return !!GLOB.syndicate_supermatter_engine
```

bobbahbrown commented on code/game/objects/items/tools/crowbar.dm:
```diff
@@ -112,3 +112,48 @@
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/crowbar/combat/attack_self(mob/living/user)
+	if(on)
+		on = FALSE
+		force = initial(force)
+		w_class = initial(w_class)
+		throwforce = initial(throwforce)
+		tool_behaviour = initial(tool_behaviour)
+		toolspeed = initial(toolspeed)
+		hitsound = "swing_hit"
+		playsound(user, 'sound/weapons/saberoff.ogg', 5, TRUE)
+		to_chat(user, "<span class='warning'>[src] is now concealed.</span>")
+	else
+		on = TRUE
+		force = 18
+		w_class = WEIGHT_CLASS_NORMAL
+		throwforce = 10
+		tool_behaviour = TOOL_CROWBAR
+		toolspeed = 0.5
+		hitsound = 'sound/weapons/blade1.ogg'
+		playsound(user, 'sound/weapons/saberon.ogg', 5, TRUE)
+		to_chat(user, "<span class='warning'>[src] is now active.</span>")
+	update_icon()
```
```suggestion
	on = !on
	force = on ? 18 : initial(force)
	w_class = on ? WEIGHT_CLASS_NORMAL : initial(w_class)
	throwforce = on ? 10 : initial(throwforce)
	tool_behaviour = on ? TOOL_CROWBAR : initial(tool_behaviour)
	toolspeed = on ? 0.5 : initial(toolspeed)
	hitsound = on ? 'sounds/weapons/blade1.ogg' : 'swing_hit'
	playsound(user, on ? 'sounds/weapons/saberon.ogg' : 'sounds/weapons/saberoff.ogg', 5, TRUE)
	to_chat(user, "<span class='warning'>[src] is now [on ? "active" : "concealed"].</span>")
	update_icon()
```

bobbahbrown commented on code/game/objects/items/tools/crowbar.dm:
```diff
@@ -112,3 +112,48 @@
+	update_icon()
+
+/obj/item/crowbar/combat/update_icon_state()
+	if(on)
+		icon_state = "[initial(icon_state)]_on"
+		item_state = "[initial(item_state)]1"
+	else
+		icon_state = "[initial(icon_state)]"
+		item_state = "[initial(item_state)]"
```
```suggestion
	icon_state = "[initial(icon_state)][on ? "_on" : ""]"
	item_state = "[initial(item_state)][on ? "1" : ""]"
```

bobbahbrown commented on code/game/objects/items/tools/wrench.dm:
```diff
@@ -103,9 +103,9 @@
 		w_class = WEIGHT_CLASS_NORMAL
-		throwforce = 8
+		throwforce = 10
 		tool_behaviour = TOOL_WRENCH
-		toolspeed = 1
+		toolspeed = 0.5
```
undocumented change, please put this in the PR body

also consider refactoring this to use ternary operators like my above suggestion

bobbahbrown commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -469,23 +474,60 @@
+/obj/effect/mob_spawn/human/syndicate/battlecruiser/engineering
+	name = "Syndicate Battlecruiser Ship Engineer"
+	short_desc = "You are an engineer aboard the syndicate flagship: the SBC Starfury."
+	flavour_text = "Your job is to maintain the ship, and keep the engine running. If you are not familiar with how the supermatter engine functions: do not attempt to start it."
```
```suggestion
	flavour_text = "Your job is to maintain the ship, and keep the engine running. If you are unfamiliar with how the supermatter engine functions do not attempt to start it alone; ask a fellow crewman for help."
```

bobbahbrown commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -469,23 +474,60 @@
+/obj/effect/mob_spawn/human/syndicate/battlecruiser/medical
+	name = "Syndicate Battlecruiser Ship Medical Doctor"
+	short_desc = "You are a medical doctor aboard the syndicate flagship: the SBC Starfury."
+	flavour_text = "Your job is to maintain crew physical health and keep your comrades alive at all cost."
```
```suggestion
	flavour_text = "Your job is to maintain the crew's physical health and keep your comrades alive at all cost."
```

bobbahbrown commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,82 @@
+	antag_hud_type = ANTAG_HUD_OPS
+	antag_hud_name = "synd"
+	antag_moodlet = /datum/mood_event/badass_antag //focused
+	var/datum/team/starfuryop/sbccrew
```
Please add documentation to make this dmdoc compliant.

EgorDinamit replied:
It's stil WIP, but what do you mean by "dmdoc compliant"?

EgorDinamit replied:
And yeah, moodlet will be "focused" instead.

bobbahbrown commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,82 @@
+			if(P.sbccrew)
+				sbccrew = P.sbccrew
+				return
+		if(!new_team)
+			sbccrew = new /datum/team/starfuryop
+			sbccrew.forge_objectives()
+			return
```
why are you checking this a second time?

EgorDinamit replied:
I actually have no idea, I'm not familiar with antag code, I'll check it all a bit later.

bobbahbrown commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,82 @@
+/datum/team/starfuryop
+	name = "Starfury battlecruiser operatives"
+
+/datum/team/starfuryop/proc/add_objective(datum/objective/O, needs_target = FALSE)
+	O.team = src
+	O.update_explanation_text()
+	objectives += O
+
+/datum/antagonist/starfuryop/proc/forge_objectives()
+	if(sbccrew)
+		objectives |= sbccrew.objectives
+
+/datum/team/starfuryop/proc/forge_objectives()
+	add_objective(new/datum/objective/syndicatesupermatter, TRUE)
```
Please add documentation to make this dmdoc compliant.

bobbahbrown commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,82 @@
+	add_objective(new/datum/objective/syndicatesupermatter, TRUE)
+
+/datum/team/starfuryop/roundend_report()
+	var/list/parts = list()
+
+	parts += "<span class='header'>The crew of syndicate battle cruiser were:</span>"
```
```suggestion
	var/list/parts = list("<span class='header'>The crew of syndicate battle cruiser were:</span>")
```

EgorDinamit replied:
I was going to add more parts to it

bobbahbrown commented on code/modules/events/starfurybc.dm:
```diff
@@ -0,0 +1,99 @@
+				announce_to_ghosts(spawner)
+
+	sleep(announcetime)
+	priority_announce("Syndicate Battle Cruiser has been found near the station's sector, brace for impact.", sound = 'sound/machines/alarm.ogg')
```
consider rewording this announcement it sounds awkward

EgorDinamit replied:
Yeah, it was just for testing, don't worry.

EgorDinamit committed:
"The"

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/game/gamemodes/objective.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/game/objects/items/tools/crowbar.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/game/objects/items/tools/crowbar.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/game/objects/structures/ghost_role_spawners.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
Update code/modules/antagonists/starfurybc/starfurybc.dm

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>

EgorDinamit committed:
' -> "

Appveyor was throwing errors about sound files, that should fix it.

AdamElTablawy commented:
This is kind of instantly round ending. It's legit gonna be more destructive than slaughter demon.

wesoda25 commented:
It should use an incinerator for power instead

Omni44 commented:
What are annexes on sides of bow of the ships meant for? the ones with some console and kinetic launchers? Are they there for aesthetic reasons or does the cruiser have anything to launch at station, other than syndies?

Acer202 commented:
Biggest complaint aside from the absolute slaughter this would be every time it comes up is the fact that the syndicate suddenly have supermatter engines. That kinda makes the shard-stealing nonsensical. Try a different engine, perhaps? In TGMC lore, ships used singularity engines for the longest time, but that's obviously not a good idea for this case. Perhaps use a generator bank, like you see in the other small-ish ships around space.

Shadowflame909 commented:
> 
> 
> This is kind of instantly round ending. It's legit gonna be more destructive than slaughter demon.

Perfect for rounds that last too long then.

What the heck are people still doing after 3 hours anyways. With 40 pop???

ATH1909 commented on code/game/objects/items/tools/crowbar.dm:
```diff
@@ -112,3 +112,33 @@
+
+/obj/item/crowbar/combat
+	name = "combat crowbar"
+	desc = "Advanced crowbar 'borrowed' from combined alliance."
```
desc = "**An a**dvanced crowbar 'borrowed' from **the** combined alliance."

EgorDinamit replied:
It's a meme description right now, but thanks.

ATH1909 commented on code/game/objects/items/tools/crowbar.dm:
```diff
@@ -112,3 +112,33 @@
+	desc = "Advanced crowbar 'borrowed' from combined alliance."
+	icon_state = "crowbar_combat"
+	item_state = "crowbar_combat"
+	attack_verb = list("devastated", "brutalized", "crowbared")
```
*crowbarred

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -469,23 +474,60 @@
-	flavour_text = "Your job is to follow your captain's orders, maintain the ship, and keep the engine running. If you are not familiar with how the supermatter engine functions: do not attempt to start it."
-	important_info = "The armory is not a candy store, and your role is not to assault the station directly, leave that work to the assault operatives."
+	flavour_text = "Your job is to follow your higher-ranking operatives' orders, assisting in pretty much anything that might need your help."
+	important_info = "While you don't have strict role, you are supposed to obey orders given by anyone on the ship, including medical, engineering and assault operatives."
```
*a strict role

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -469,23 +474,60 @@
 
 /obj/effect/mob_spawn/human/syndicate/battlecruiser
 	name = "Syndicate Battlecruiser Ship Operative"
 	short_desc = "You are a crewmember aboard the syndicate flagship: the SBC Starfury."
```
should "syndicate" be capitalized (here and elsewhere)?

EgorDinamit replied:
Ah, you mean in description? Makes sense, I'll change it.

ATH1909 commented on code/game/objects/structures/ghost_role_spawners.dm:
```diff
@@ -469,23 +474,60 @@
+/obj/effect/mob_spawn/human/syndicate/battlecruiser/engineering
+	name = "Syndicate Battlecruiser Ship Engineer"
+	short_desc = "You are an engineer aboard the syndicate flagship: the SBC Starfury."
+	flavour_text = "Your job is to maintain the ship, and keep the engine running. If you are unfamiliar with how the supermatter engine functions do not attempt to start it alone; ask a fellow crewman for help."
```
add a comma after "functions"

ATH1909 commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,80 @@
+			win = FALSE
+		objective_count++
+	if(win)
+		parts += "<span class='greentext'>Crew of syndicate battle cruiser were successful!</span>"
```
add a "the" before "syndicate"

ATH1909 commented on code/modules/antagonists/starfurybc/starfurybc.dm:
```diff
@@ -0,0 +1,80 @@
+	if(win)
+		parts += "<span class='greentext'>Crew of syndicate battle cruiser were successful!</span>"
+	else
+		parts += "<span class='redtext'>Crew of syndicate battle cruiser have failed!</span>"
```
same here

ATH1909 commented:
The Syndicate doctors and engineers not being allowed to join in on the big murderbone assault is kind of lame. Let them join in on the fun, I say.

EgorDinamit commented:
> The Syndicate doctors and engineers not being allowed to join in on the big murderbone assault is kind of lame. Let them join in on the fun, I say.

They are allowed, but in my opinion not all of them should be on station full-time.
Doctors get med-beam for a reason, but it's also good if one will stay on the ship to heal dead allies.
Engies can easily hack everything as well as fix the ship in case of bombing from the enemies.

EgorDinamit commented:
> Biggest complaint aside from the absolute slaughter this would be every time it comes up is the fact that the syndicate suddenly have supermatter engines. That kinda makes the shard-stealing nonsensical. Try a different engine, perhaps? In TGMC lore, ships used singularity engines for the longest time, but that's obviously not a good idea for this case. Perhaps use a generator bank, like you see in the other small-ish ships around space.

Almost the only reason for using SM is objective.
If SM is destroyed - Syndicate already lost this battle, as in lore-wise they don't have a lot of supermatter crystals(imo it's ~10).

EgorDinamit committed:
minor changes

Chaznoodles commented:
Isn't this just the equivalent of nukeops firing on dynamic though

itseasytosee commented:
In hindsight, this just seems like a way to end rounds that go on for 2+ hours

necromanceranne commented:
What the fuck is even the point of any other hour deep midrounds when this shit is the be all, end all of midrounds as it singlehandedly forces a shuttle call and wipes the remaining crew off the map. I don't even have adequate words to describe how horrendously bad spawning a team of syndicate death commandos complete with maulers and the highest damaging weapons in the game, and launching them at the station truly is. These fuckers are more numerous and better equipped than ops of an equivalent population, and they don't have an objective other than kill and maim. This is basically the last thing tg needs for content.

EgorDinamit commented on code/game/objects/items/tools/wrench.dm:
```diff
@@ -90,30 +90,16 @@
-		playsound(user, 'sound/weapons/saberon.ogg', 5, TRUE)
-		to_chat(user, "<span class='warning'>[src] is now active. Woe onto your enemies!</span>")
+	on = !on
+	force = on ? 8 : initial(force)
```
Oh, wait, energy wrench is actually used somewhere? I did it because I thought it's not used anywhere, so engies on cruiser will have some sort of a gimmick weapon.

EgorDinamit commented on code/game/objects/items/storage/belt.dm:
```diff
@@ -115,7 +115,7 @@
 	new /obj/item/wrench/combat(src)
 	new /obj/item/weldingtool/largetank(src)
-	new /obj/item/crowbar(src)
+	new /obj/item/crowbar/combat(src)
```
Sorry, I thought it's not actually used anywhere, I'll make another toolbelt type for that.

EgorDinamit commented:
@necromanceranne thanks for pointing out, I didn't know that energy wrench is actually used anywhere.
@itseasytosee I'd say it's not a way to simply finish the round, it's the way to give a proper challenge once boredom kicks in. 
Most of 1+ hours rounds finish with "It's boring, call shuttle? Y/N".
The whole point here is that we'd have 13 syndies AND at least 26 crew members on the station, and given the tech, you MUST have at 1+ hour mark, it's a fair fight, I'd even say syndies here are weaker than the crew with all mechs and advanced weaponry.

EgorDinamit committed:
Update belt.dm

EgorDinamit committed:
Update wrench.dm

Energy wrench should be a good tool anyway, but let's not give them 0.5 toolspeed.

itseasytosee commented:
Hey, do you know about this really neat gamemode called nuclear operatives where a small team of syndies with advanced equipment attempts to blow up the station? It real neat and I think you should research or at least recognize some of the balance implications. The purpose of the station is not to prepare for a giant bosses battle.
In my eyes, this is a clear example of unga make thing because it sound cool and not care about balance implications or the fun of others.

Shadowflame909 commented:
Yeah well have you heard of this really awesome thing called content. After 1 hour at worst it's all gone for a job. 

People are just going brain-dead after that mark so a good forced crew transfer to refresh the station is necessary.

In other news, this'll actually cause mining, botany, xenobio, virology, and robotics powergame to actually have a pay off

Fikou commented:
this event can spawn at 1 hour (which isnt that far for the crew, they reach endgame at about 1.5-2 hours), and has 8 weight, it spawns 13 fucking syndicate operatives with the goal to murderbone, 
 and nothing else except protect a supermatter crystal, they can murderbone with l6 saws, laser guns, fucking maulers and other overpowered nukie equipment, they get 3 emags, turrets, thermals and eswords, all for free, without needing any uplink even though there's a lot more of them than normal nukies, they get survival pens for free heals and self use sleepers, combat defibs and medhealguns, the admiral has night vision thermal sec hudglasses (lol)
nukies have a limited amount of tc, have less people, have a harder objective and if they choose the path to have fun (war ops) they need to wait 20 minutes while the crew gears up while this has 8, and the station will likely be pretty damaged a few hours in
a slaughter demon for random event has 1 hour, 1 weight making it extremely rare, and its 1 simple mob that has big force and can teleport, thats all
this event is horribly unbalanced and it sounds like a replacement for nukies or a cool admin event instead of a random event that can trigger an hour in

EgorDinamit commented:
@Fikou they don't get L6, lol.
I'm not giving them uplinks so they can't buy said L6 and other super overpowered stuff, they have to work around with equipment they have.
Pirates event has weight of 8, in theory it means you would probably see THIS event at 1:30 - 2:00 mark, not at 1 hour, though I'd probably lower it to 4-6.

I'd probably change some checks needed for event to start too: Instead of 40 total pop it would instead check if there are at least 40 LIVING crewmembers, not including ghosts and ghost roles.

Fikou commented:
> @Fikou they don't get L6, lol.
> I'm not giving them uplinks so they can't buy said L6 and other super overpowered stuff, they have to work around with equipment they have.
> Pirates event has weight of 8, in theory it means you would probably see THIS event at 1:30 - 2:00 mark, not at 1 hour, though I'd probably lower it to 4-6.
> 
> I'd probably change some checks needed for event to start too: Instead of 40 total pop it would instead check if there are at least 40 LIVING crewmembers, not including ghosts and ghost roles.

oh its m90's not l6 saws, those are still 1 shot unarmored peope though, they still have super overpowered stuff
the problem with this is that this WILL force a shuttle call, other antags are destructive but they can usually be dealt with since they are a single force, this is 13 overpowered guys that come to the station to destroy it, which will be insta called just like nukies, you can even completely skip over the event by red alert calling the shuttle when the announcement appears

EgorDinamit commented:
@Fikou once event begins, shuttle is delayed by 14 minutes.
I'm still trying to balance it to point "You have gear, but not everyone will have it"
I'm still thinking about how much people can do at 1-2 hours mark.

EgorDinamit commented:
@Fikou once event begins, shuttle is delayed by 14 minutes.
I am sill thinking about how much people can do at 1-2 hours mark.

EgorDinamit commented:
@Fikou once event begins, shuttle is delayed by 14 minutes.
I am sill thinking about how much people can do at 1-2 hours mark.

itseasytosee commented:
I really don't see what this artificially other than artificially ending a round.

EgorDinamit commented:
It puts a use for all late tech you have at this point.
All those mechs you can make, all those weapons you can print in a moment, all those bombs you can make - everything will be finally useful for something that is not antag-murderboning, fighting against war ops(arguably because 30 minutes shift) or xenos.

itseasytosee commented:
Why would you want to incentivize stockpiling bombs and weapons? There are a lot less researchable weapons than you think and they already have their uses. Xray gun and marksman rifles are used for blob and ion rifles are used for Anti AI and the tesla revolver and temp gun are pretty useless anyway. The Advanced Energy Gun is the only printable weapon you could maybe make a case for. Bombs already have uses (and look like their gonna have more soon.) All of those things either already HAVE a use or are useless in the situation.

Other midround antags like pirate and swarmer are meant to be an overcomeable yet moderate threat. The only ones that can end the round are when blobs and Xenos get out of control.

EgorDinamit commented:
Well, I want it to be something close to xenos then.
You get time to prepare, you can(will) see location of the cruiser on gps, you can win against 13 people with a team of at least 26.

itseasytosee commented:
The reason single antags can kill many people is a matter of lack of organization. A traitor with a solid murder based kit can kill person after person but would be much more difficult to fight six or even three people at once. if nukies declare war and everyone guards the cap then its a piece of cake disregarding crowd control item. This rarely happens but it's a similar concept to the action economy in dnd 5e due to how fights work and how fights end by one side stunning or criting another. What I'm getting at is these guys will steamroll and it's not gonna be a very fun round when they win. What's their goal anyway? what happens when they kill everyone on the station? just twiddle their thumbs till centcomm sends enough death squads to call the shuttle to end the round.

EgorDinamit commented:
Again, Work In Progress.
I wanted to automatically spawn ERT team once 50% of the crew dies on the same z-level as Syndicate Cruiser to destroy said cruiser and then probably get rid of remaining operatives and call the shuttle.
Another idea was just to force Delta/Red alert shuttle call without possibility to recall, but that's no _that_ interesting.

optimumtact commented:
`The event can trigger only after 1 hour and only if the server has more than 40 active players.`

It's way to early for this, this is a 2 hour event at earliest imo

TheSilverNuke commented:
I would also reduce the number of spawns to about 6, especially considering what @itseasytosee mentioned about the lack of organization between crew. Six operatives is a LOT, thirteen would just be too many. I approve of this PR but it should be balanced to be fun as well.

EgorDinamit commented:
@optimumtact realistically, you wouldn't see a lot of 2+ hours shifts, and given the chance, it'd rarely spawn exactly at 1 hour mark.

EgorDinamit commented:
@TheSilverNuke understandable, but I really want them to besomewhat equal to end-game crew. 
13 is indeed too much, at least because you wouldn't be able to find such amount of ghosts.

optimumtact commented:
Realistically, you shouldn't be seeing this event in many rounds, so that's entirely in keeping with the point.

This is a round ending random event in pretty much every way, so by your own hand you have consigned it to irrelevance.

This really goes against the whole spirit of the random event, which is something which should have some impact on the round, but not materially impact it's actual pacing and out come.

Frankly the more I think about this the worse it seems, so I'll be closing this.

This should be a gamemode, not an event.

EgorDinamit commented:
So, time to rework it into event then.

itseasytosee commented:
Nukies with fewer options and more people sounds like a real shit gamemode to me.
